File: example/back-end/ApiBackEnd/ViewModels/BasicUploadViewModel.cs
using System.ComponentModel.DataAnnotations;
using ApiBackEnd.Models;
using ApiMultiPartFormData.Models;

namespace ApiBackEnd.ViewModels
{
    public class BasicUploadViewModel
    {
        #region Properties

        /// <summary>
        /// Author information.
        /// </summary>
        [Required]
        public User Author { get; set; }

        /// <summary>
        /// Attachment.
        /// </summary>
        [Required]
        public HttpFile Attachment { get; set; }

        #endregion
    }
}

File: example/front-end/ngx-example/src/view-models/basic-upload.view-model.ts
import {User} from '../models/user';

export class BasicUploadViewModel {
  //#region Properties

  /*
  * User information.
  * */
  public user: User;

  /*
  * Attachment to upload.
  * */
  public attachment: File;

  //#endregion

  //#region Constructor

  public constructor(){
    this.user = new User();
  }

  //#endregion
}
